### PR TITLE
Fix #9585 - WNP 83 (Code) 

### LINKED ISSUE #9585: WNP 83: Pocket - Code
Epic Issue: #9573

Success: 

- [ ] Review site
- [ ] Localized to X languages (TBD)

### PR DESCRIPTION
**Pull request by maxxcrawford**
## Description

Add WNP 83 Page

## Issue / Bugzilla link

#9585 (Code task) 
#9573 (Epic) 

TODO: 

- [x] Optimize SVGs
- [x] Remove unused code (SCSS, JS)
- [x] Confirm final copy 
- [x] Update Pocket logo to have dark/light theme

## Testing

URLs: 
- Local: http://localhost:8000/en-US/firefox/83.0/whatsnew/all/
- PR/Dev: https://www-demo-pr-9588.herokuapp.com/en-US/firefox/83.0/whatsnew/all/

Assets:
- [ ]  Confirm design: https://www.figma.com/file/UwstJV4VVSQgIaH9ZP7xGm/Firefox-WNP?node-id=63%3A39422
- [ ] Confirm copy: https://docs.google.com/document/d/1zo6TYWw0rdYET-c9vskuGKPjEktvSIM5kExugTFJ_Kc/edit?usp=sharing

### CODE REVIEW

File: bedrock/firefox/tests/test_base.py [hunk 1]
```diff
@@ -375,6 +375,19 @@ def test_fx_82_0_0_china(self, render_mock):
 
     # end 82.0 whatsnew tests
 
+    # begin 83.0 whatsnew tests
+
```
Comment (maxxcrawford) [inline bedrock/firefox/tests/test_base.py:379]: We're going to localize this WNP for a specific set of languages (that match the same l10n from Pocket). Should we test the routing works for those specific languages?

File: bedrock/firefox/templates/firefox/whatsnew/includes/fx83/pocket.svg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<svg width="127" height="115" viewBox="0 0 127 115" xmlns="http://www.w3.org/2000/svg">
```
Comment (maxxcrawford) [inline bedrock/firefox/templates/firefox/whatsnew/includes/fx83/pocket.svg:1]: We're using THIS Pocket file because the Pocket SVG in Protocol doesn't have the white background for the downward-facing caret. 
image

File: bedrock/firefox/views.py [hunk 3]
```diff
@@ -612,6 +613,8 @@ def get_template_names(self):
                 template = 'firefox/developer/whatsnew.html'
             else:
                 template = 'firefox/whatsnew/index.html'
+        elif version.startswith('83.') and ftl_file_is_active('firefox/whatsnew/whatsnew-fx83'):
+            template = 'firefox/whatsnew/whatsnew-fx83.html'
```
Comment (maxxcrawford) [inline bedrock/firefox/views.py:617]: Should this have the same sort of `locale.startswith('en-'):` logic for the supported languages from WNP82?

File: l10n/en/firefox/whatsnew/whatsnew-fx83.ftl [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/firefox/83.0/whatsnew/all/
+
+# html page title
+whatsnew83-page-title = What’s new with { -brand-name-firefox }
+
+# Main title
+whatsnew83-main-title = Meet { -brand-name-pocket } &mdash; Our inspired source for news & stories
```
Comment (maxxcrawford) [inline l10n/en/firefox/whatsnew/whatsnew-fx83.ftl:11]: Is `&mdash;` okay in this FTL file?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx82.html [hunk 5]
```diff
@@ -40,7 +40,7 @@ <h1 class="wnp-main-title">Firefox protects your browsing. <strong>Mozilla VPN</
 
         <div class="c-picto-block">
           <div class="c-picto-block-image">
-            {% include 'firefox/whatsnew/includes/illo-vpn-laptop.svg' %}
```
Comment (maxxcrawford) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx82.html:43]: Renamed these images to match the new convention for the inline dark/light mode SVGs. 
- `bedrock/firefox/templates/firefox/whatsnew/includes/fx82/illo-vpn-mobile.svg`
- `bedrock/firefox/templates/firefox/whatsnew/includes/fx82/illo-vpn-laptop.svg`

File: bedrock/firefox/tests/test_base.py (hunk 1 above)
Comment (stephaniehobson) [reply bedrock/firefox/tests/test_base.py:379]: How long is the list? If it's 3, test them all. If it's 20 I think it's enough to test EN and one other local expected to display 83 and one other expected not too.

File: bedrock/firefox/templates/firefox/whatsnew/includes/fx83/pocket.svg (hunk 2 above)
Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/whatsnew/includes/fx83/pocket.svg:1]: 👍

File: bedrock/firefox/views.py (hunk 3 above)
Comment (stephaniehobson) [reply bedrock/firefox/views.py:617]: IMHO the `ftl_file_is_active` is enough of a check.

File: bedrock/firefox/tests/test_base.py (hunk 1 above)
Comment (stephaniehobson) [reply bedrock/firefox/tests/test_base.py:379]: Oh, the list of which are supported and not is managed by l10n. I don't think you can write a test for this.... hold please.

Comment (stephaniehobson) [reply bedrock/firefox/tests/test_base.py:379]: Okay, I think what you want to do is write a test for `en-US`, and then one for a different local, like `de` and test it with `ftl_file_is_active` set to `True` and to `False` to check the fallback works as expected.

Comment (stephaniehobson) [reply bedrock/firefox/tests/test_base.py:379]: Or you can copy the test for 81 or 80, since I think those pages show up under the same conditions.

File: l10n/en/firefox/whatsnew/whatsnew-fx83.ftl (hunk 4 above)
Comment (stephaniehobson) [reply l10n/en/firefox/whatsnew/whatsnew-fx83.ftl:11]: Yes. It would be worth adding a comment to explain its usage though. (single-line comments in FTL files are indicated with a single #)

`# The — or mdash is used here like a colon :`

File: bedrock/firefox/tests/test_base.py (hunk 1 above)
Comment (stephaniehobson) [reply bedrock/firefox/tests/test_base.py:379]: @pmac Am I right about this? Can we specify if the language file is active [in a test like this](https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/tests/test_base.py#L336).

File: bedrock/firefox/views.py (hunk 3 above)
Comment (stephaniehobson) [reply bedrock/firefox/views.py:617]: Is there a designated someone who will let Peying know which locales need to be activated?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  {% include 'firefox/whatsnew/includes/header.html' %}
+
+  <section class="wnp-content-main">
+    <div class="mzp-l-content mzp-t-content-md wnp83-hero-logo">
+      {% include 'firefox/whatsnew/includes/fx83/pocket.svg' %}
+      <h1 class="wnp-main-title">{{ ftl('whatsnew83-main-title') }}</h1>
+      <p class="wnp-main-tagline">{{ ftl('whatsnew83-main-intro') }}</p>
+
+      <div class="wnp-main-cta">
+        <a href="https://getpocket.com/explore?utm_source=mozilla.org-whatsnew83&utm_medium=referral&utm_campaign=whatsnew83&entrypoint=mozilla.org-whatsnew83" class="mzp-c-button mzp-t-product mzp-t-lg" rel="external" data-cta-text="Get Mozilla VPN Now" data-cta-type="button">{{ ftl('whatsnew83-main-cta') }}</a>
```
Comment (stephaniehobson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html:32]:
```suggestion
        <a href="https://getpocket.com/explore?utm_source=mozilla.org-whatsnew83&utm_medium=referral&utm_campaign=whatsnew83&entrypoint=mozilla.org-whatsnew83" class="mzp-c-button mzp-t-product mzp-t-lg" rel="external" data-cta-text="Get Pocket Now" data-cta-type="button">{{ ftl('whatsnew83-main-cta') }}</a>
```

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+  <section class="wnp-content-extra">
+    <div class="mzp-l-content mzp-t-content-xl">
+      <div class="l-columns-three">
+
+        <div class="c-picto-block">
+          <div class="c-picto-block-image">
+            {% include 'firefox/whatsnew/includes/fx83/person1.svg' %}
+          </div>
+          <h3 class="c-picto-block-title">{{ ftl('whatsnew83-quality-reads') }}</h3>
```
Comment (stephaniehobson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html:45]: These 3 headings should be `h2`s to preserve the heading hierarchy.

File: media/static-bundles.json [hunk 8]
```diff
@@ -843,6 +850,15 @@
       ],
       "name": "firefox_whatsnew_82"
     },
+    {
+      "files": [
+        "js/firefox/whatsnew/up-to-date.js",
+        "js/libs/spin.min.js",
+        "js/base/send-to-device.js",
```
Comment (stephaniehobson) [inline media/static-bundles.json:857]: I don't see the send to device widget in the HTML. Is this the code you said you need to remove?

**Review by stephaniehobson — CHANGES_REQUESTED** [review]
Couple nit picks but overall it's looking good.

File: bedrock/firefox/views.py (hunk 3 above)
Comment (pmac) [reply bedrock/firefox/views.py:617]: `ftl_file_is_active` is always `True` for English locales.

File: bedrock/firefox/tests/test_base.py (hunk 1 above)
Comment (pmac) [reply bedrock/firefox/tests/test_base.py:379]: Do you mean you just want to test that our localization system works for this page? If so I don't think we need to do that. We have tests that prove that pages can be activated in languages and that l10n works. If there is locale specific template choosing in the view then that can be tested of course. Is that what we're saying? From the view code it looks like all we care about is that the Fluent file is active, so I don't think a test is needed beyond what @stephaniehobson suggested with th emock of `ftl_file_is_active` and making sure it does what you expect when that returns `True` and `False`.

File: media/css/firefox/whatsnew/whatsnew-83.scss [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    margin: 0 auto $spacing-2xl;
+    max-width: $content-md - ($layout-md * 2);
+    padding: 0 $layout-md;
+
+    .c-picto-block-title {
+        @include text-title-xs;
+    }
+
+    .c-picto-block-image {
+        height: 250px;
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply media/css/firefox/whatsnew/whatsnew-83.scss:72]: 🔠

[the remark this answers is not in the document]
Comment (maxxcrawford) [reply media/css/firefox/whatsnew/whatsnew-83.scss:72]: FWIW, I've never done this, but love the organization. Adding this to my mental toolkit.

File: media/css/firefox/whatsnew/whatsnew-83.scss [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+    @include text-body-sm;
+    max-width: $content-md;
+    padding-bottom: $layout-xl;
+    text-align: center;
+}
+
+
+//* -------------------------------------------------------------------------- */
+// Footer overrides
+.mzp-c-footer.mzp-t-firefox {
```
[the remark this answers is not in the document]
Comment (maxxcrawford) [reply media/css/firefox/whatsnew/whatsnew-83.scss:124]: @stephaniehobson Opened up an issue for this per the convo on Slack. https://github.com/mozilla/bedrock/issues/9629

File: media/static-bundles.json (hunk 8 above)
Comment (maxxcrawford) [reply media/static-bundles.json:863]: I left it as it was in 82 and was the same. Wasn't sure if there was a fallback view or something. 

@craigcook Am I clear to remove this? I can also remove some references in the SCSS for it too. _(Both the 82 JS file and the SCSS file were copied over to 83 as starting points)_

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html (hunk 6 above)
Comment (maxxcrawford) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html:32]: Do I need to use the `ftl` entry, or can it be in plain English?

Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html]: Plain English for these. 

When GTM takes these and puts them in GA this value becomes the event label. Setting this value in a data attribute means two things: 1) we can tweak the text without affecting the tracking 2) we can compare performance across locales using the same event labels.

(I think it's event lable - could be event action)

Comment (maxxcrawford) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx83.html]: 👍

File: media/static-bundles.json (hunk 8 above)
Comment (craigcook) [reply media/static-bundles.json:863]: 82 also doesn't have the widget, so this is redundant in both bundles. I would have copied from 81 which *does* have the widget, and just neglected to remove it (and was also missed in review). We should remove send-to-device.js and spin-min.js from both bundles.

File: bedrock/firefox/views.py [hunk 11]
```diff
@@ -649,7 +652,7 @@ def get_template_names(self):
 class WhatsNewChinaView(WhatsnewView):
     def get_template_names(self):
         template = super().get_template_names()
-        if template == ['firefox/whatsnew/whatsnew-fx82.html']:
+        if template == ['firefox/whatsnew/whatsnew-fx83.html']:
```
Comment (craigcook) [inline bedrock/firefox/views.py:655]: This shouldn't be changed. The special view for China was specific to WNP82 since that page promoted VPN, which we can't promote in China, we don't need to update it for 83 (and when WNP82 is eventually deleted we won't need this view at all).

File: media/static-bundles.json (hunk 8 above)
Comment (maxxcrawford) [reply media/static-bundles.json:863]: Haha. Thank you both. Removing!

File: bedrock/firefox/templates/firefox/whatsnew/includes/fx83/person2.svg [hunk 12]
```diff
@@ -0,0 +1 @@
...2-2.852 1.996-2.068 3.565l14.831 32.371a3.602 3.602 0 003.28 2.139l37.648.999c3.494.285 6.703-1.925 7.772-5.348z" fill="#fff"/><path d="M104.5 194.066c2.639 0 4.778-1.852 4.778-4.136s-2.139-4.135-4.778-4.135c-2.638 0-4.777 1.851-4.777 4.135s2.139 4.136 4.777 4.136z" stroke="#000" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/><path d="M47.386 155.134s11.48-2.353 17.113-6.203c5.633-3.851 17.256-15.33 18.254-16.685.998-1.355-8.556 1.355-9.127 1.569-.57.214-19.252 12.62-19.252 12.62l-6.988 8.699z" fill="#FCB643"/><path d="M42.68 156.132c.57 0 16.543-2.923 26.026-10.481s17.47-16.685 19.822-22.532c2.353-5.847 5.277-12.621 7.558-14.974 2.282-2.353 4.493-3.137 4.493-3.137M106.212 101.586s4.492-.571 6.773 2.139c2.282 2.709 5.277 8.913 6.418 12.05 1.14 3.137 2.923 6.631 7.415 7.772" stroke="#000" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/><path d="M96.728 103.511s-4.492.998-9.126 6.417c-3.138 3.708-12.407 20.036-20.037 25.812-7.63 5.775-13.619 9.911-18.824 11.266-5.276 1.354-13.048 0-13.048 0" stroke="#000" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (stephaniehobson) [inline bedrock/firefox/templates/firefox/whatsnew/includes/fx83/person2.svg:1]: The tool you used to minimize these removed the view box. That's okay for WNP since they're Firefox only but you should adjust the settings so it doesn't do this if the image is going to be displayed on other pages.

File: media/static-bundles.json [hunk 13]
```diff
@@ -842,13 +847,16 @@
     },
     {
       "files": [
-        "js/firefox/whatsnew/up-to-date.js",
-        "js/libs/spin.min.js",
-        "js/base/send-to-device.js",
-        "js/firefox/whatsnew/whatsnew-82.js"
+        "js/firefox/whatsnew/up-to-date.js"
```
Comment (stephaniehobson) [inline media/static-bundles.json:850]: I don't think you meant to edit the 82 JS file?

**Review by stephaniehobson — CHANGES_REQUESTED** [review]
So close, just one question about the static bundle for 82.

File: media/static-bundles.json (hunk 13 above)
Comment (maxxcrawford) [reply media/static-bundles.json:850]: I did – per @craigcook's [comment](https://github.com/mozilla/bedrock/pull/9588#discussion_r514487587) that 82 also wasn't using the `send-to-device`/`spin` files.

File: bedrock/firefox/templates/firefox/whatsnew/includes/fx83/person2.svg (hunk 12 above)
Comment (maxxcrawford) [reply bedrock/firefox/templates/firefox/whatsnew/includes/fx83/person2.svg:1]: I'll reformat these to keep the view box.

Comment (maxxcrawford) [reply bedrock/firefox/templates/firefox/whatsnew/includes/fx83/person2.svg]: Done!

**Review by stephaniehobson — APPROVED** [review]
R+ 🌚
